cmd/sourced/compose/workdir/factory.go

```diff
@@ -143,78 +136,137 @@ func initialize(dirName string, subPath string, envf envFile) (*Workdir, error)
 }
 
 type envFile struct {
-	Workdir             string
-	ReposDir            string
+	ComposeProjectName string
+
+	GitbaseVolumeType   string
+	GitbaseVolumeSource string
+	GitbaseSiva         bool
+
 	GithubOrganizations []string
 	GithubToken         string
-	WithForks           bool
+
+	NoForks bool
+
+	GitbaseLimitCPU      float32
+	GitcollectorLimitCPU float32
+	GitbaseLimitMem      uint64
 }
 
-func (f *envFile) MarshalEnv() ([]byte, error) {
-	volumeType := "bind"
-	volumeSource := f.ReposDir
-	gitbaseSiva := ""
-	if f.ReposDir == "" {
-		volumeType = "volume"
-		volumeSource = "gitbase_repositories"
-		gitbaseSiva = "true"
+func newLocalEnvFile(dirName, repoDir string) envFile {
+	f := envFile{
+		ComposeProjectName: fmt.Sprintf("srcd-%s", dirName),
+
+		GitbaseVolumeType:   "bind",
+		GitbaseVolumeSource: repoDir,
 	}
+	f.addResourceLimits()
+
+	return f
+}
 
-	noForks := "true"
-	if f.WithForks {
-		noForks = "false"
+func newOrgEnvFile(dirName string, orgs []string, token string, withForks bool) envFile {
+	f := envFile{
+		ComposeProjectName: fmt.Sprintf("srcd-%s", dirName),
+
+		GitbaseVolumeType:   "volume",
+		GitbaseVolumeSource: "gitbase_repositories",
+		GitbaseSiva:         true,
+
+		GithubOrganizations: orgs,
+		GithubToken:         token,
+
+		NoForks: !withForks,
 	}
+	f.addResourceLimits()
 
+	return f
+}
+
+func (f *envFile) addResourceLimits() {
 	// limit CPU for containers
-	gitbaseLimitCPU := "0.0"
-	gitcollectorLimitCPU := "0.0"
 	dockerCPUs, err := dockerNumCPU()
 	if err != nil { // show warning
 		fmt.Println(err)
 	}
 	// apply gitbase resource limits only when docker runs without any global limits
 	// it's default behaviour on linux
 	if runtime.NumCPU() == dockerCPUs {
-		gitbaseLimitCPU = fmt.Sprintf("%.1f", float32(dockerCPUs)-0.1)
+		f.GitbaseLimitCPU = float32(dockerCPUs) - 0.1
 	}
+	// always apply gitcollector limit
 	if dockerCPUs > 0 {
```

Comment:
When this can be `<= 0`?

Reply:
It shouldn't be `< 0` ever (unless something went really wrong) but it is `0` when for some reason we couldn't get info from docker.  If we don't know how many CPUs available to the docker we can't set any limit.